### PR TITLE
Update 04-troubleshooting.md

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

Add the spring clean option to troubleshooting

## What is the value of this and can you measure success?

Developer productivity through the roof.

### CODE REVIEW

**Comment by gtrufitt** [comment]
@guardian/dotcom-platform

File: docs/01-start-here/04-troubleshooting.md [hunk 1]
```diff
@@ -2,6 +2,33 @@
...
+[dev-build] clean
+```
+
+And then running `compile`:
+
+```
+[dev-build] compile
+```
+
+If you're still seeing errors, try clearing out your target folders:
```
Comment (SiAdcock) [inline docs/01-start-here/04-troubleshooting.md]: Maybe before this, we could recommend running the mighty `cleanAll` command at `root`?

Comment (SiAdcock) [reply docs/01-start-here/04-troubleshooting.md]: Not sure, I thought `cleanAll` was more all-encompassing - does it clean all projects? Might be worth asking a server side person 😝

Comment (AWare) [reply docs/01-start-here/04-troubleshooting.md]: `git clean -fxd` is the strongest cleaner here

Comment (nicl) [reply docs/01-start-here/04-troubleshooting.md]: Re the cleanAll stuff, yep it cleans all of the Scala SBT projects. The cost is that you have to recompile everything/whatever you are working on fully next time you run it (can be up to 10 mins for dev-build) but sometimes it's needed!

Comment (SiAdcock) [reply docs/01-start-here/04-troubleshooting.md]: Okay, given that following a cleanAll and deleting targets the application takes an age to recompile, I'd say this is the order we should try things in:

- `clean` for individual project
- `cleanAll` at root
- `git clean -fxd` (deletes all ignored code, i.e. `target/` directories, etc?)

Does everyone agree?

Comment (gtrufitt) [reply docs/01-start-here/04-troubleshooting.md]: Agree.

Have updated.

**Review by SiAdcock — APPROVED** [review]
Great! I predict I'll be sharing a link to this readme _a lot_

**Comment by gtrufitt** [comment]
> Great! I predict I'll be sharing a link to this readme _a lot_

Ha... yuppppp...

**Pull request merged**